typescript/base.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 OpenStax Poland
+# Licensed under the MIT license. See LICENSE file in the project root for
+# full license text.
+
+parser: '@typescript-eslint/parser'
+
+extends:
+  - ../base.yml
+  - plugin:import/typescript
+  - plugin:@typescript-eslint/recommended
+
+rules:
+  # Forbid use of the any type.
+  '@typescript-eslint/no-explicit-any': error
```

Comment:
I would be OK with enabling `ignoreRestArgs` for this rule, but otherwise I don't see why it should be allowed.

Reply:
`any` is often used in the development and setting this to an `error` will not allow you use this shorthand.
Maybe we should have separate configs for development and production? I feel like I would agree with setting `error` if this would be only for a build step

Reply:
Nothing stops you from ignoring this error during development. 

When deciding which lint should be a warning and which an error my approach was as follows: errors must never appear on the master branch, warnings may but should be carefully reviewed.

Reply:
> Nothing stops you from ignoring this error during development.

I mean compilers will not compile code automatically if there are errors (when started in watch mode) so you can't just write the code > wait for app to refresh > check if it works, but I think there should be an option to disabled this during development directly in the project

> When deciding which lint should be a warning and which an error my approach was as follows: errors must never appear on the master branch, warnings may but should be carefully reviewed.

Okay, I agree with that statement

About this particular lint rule: I don't think it should be an error. I don't see a reason to write code like:
```
function hasArrayProp(arg: unknown) {
  return typeof arg === 'object'
    && (arg as { [key: string]: unknown })
    && (arg as { [key: string]: unknown }).arrayProp
    && Array.isArray((arg as { [key: string]: unknown }).arrayProp)
}
```

If I can write it like:
```
function hasArrayProp(arg: any) {
  return arg && Array.isArray(arg.arrayProp)
}
```